### PR TITLE
Improved handling of reference data files and removed sorting of data

### PR DESCRIPTION
**Pull request by ericchansen**
Removed sorting and improved importing reference data.

TL;DR The order you give filename arguments to calculate.py
      matters now!

Added mechanism to import reference data from a file in
commit 6d9f729492d0b942. This relied upon a 3 column format
where column 1 was a label, 2 was a weight and 3 was the
value of the data point.

The label was broken down to ascribe the various Datum
attributes. However, the method wasn't robust enough and
would fail for certain filenames. To account for all
filenames would be virtually impossible.

This is a problem because that extracted data is necessary
for the data sorting algorithms to work. More or less,
this means that with the current system the data from
reference data files will almost always be sorted
improperly.

There are really only two solutions:

 1. The reference data file will have to become a massive
    .csv file. Each row could be a data point, and there
    would be a column for each Datum attribute. This would
    be a huge burdon for the user because so many columns
    would have to be filled in with descriptions (or they
    would at least have to write None many times over).

 2. Don't rely on sorting anymore. I'm going with this
    option. It's not that much more work for the user. It
    will also avoid any problems with the sorting algorithm
    that occur with odd filenames, and it could make the
    execution of the code more efficient.

Changes to make this work:

 - calculate.py will now add data types to the overall data
   list in the same order every time.
 - The order that you give filename arguments to
   calculate.py WILL MATTER NOW.
 - It's your responsibility to ensure that the data in the
   reference data file is aligned with the data that
   calculate.py generates for the FF.

### CODE REVIEW

**Comment by arosale4** [comment]
I have been testing this prior to this merge for about 2-3 weeks. The features that use Gaussian, and macromodel data seem to work as expected to.
